rules/api/serializers.py

```diff
@@ -182,3 +183,19 @@ class Meta:
     @staticmethod
     def get_username(request):
         return request.user.username
+
+
+class UserEngagementSerializer(serializers.ModelSerializer):
+
+    engagements = serializers.SerializerMethodField()
+
+    class Meta:
+        model = get_user_model()
+        fields = ('username', 'engagements')
+
+    def get_engagements(self, obj):
+        engs = obj.engagements.all()
+        user_tags = []
+        for eng in engs:
```

Comment:
If you are going down that road, take a look at using [`values_list`](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#values-list) instead of a for loop.

Reply:
I’ve tried values_list but then there are no more 2 queries :)